internal/controller/envoy_gateway_extension_reconciler.go

```diff
@@ -206,21 +206,29 @@ func (r *EnvoyGatewayExtensionReconciler) buildWasmConfigs(ctx context.Context,
 		// rate limit
 		if effectivePolicy, ok := effectiveRateLimitPoliciesMap[pathID]; ok {
 			rlAction := buildWasmActionsForRateLimit(effectivePolicy, isRateLimitPolicyAcceptedAndNotDeletedFunc(state))
-			if hasAuthAccess(rlAction) {
-				actions = append(actions, rlAction...)
+			mergedRlActions, err := mergeAndVerify(rlAction)
```

Comment:
have you considered merging after both RLP and TRLP actions have been added? Would it work? The advantage I see is that you could merge actions from both type of policies altogether as well. Why not?